src/com/SocketCommunication.cpp

```diff
@@ -267,20 +243,14 @@ void SocketCommunication::requestConnectionAsClient(std::string      const &acce
   
   for (auto const & acceptorRank : acceptorRanks) {
     _isConnected = false;
-    std::string address;
-    const std::string addressFileName("." + requesterName + "-" +
-                                      acceptorName + "-" + std::to_string(acceptorRank) + ".address");
+    ConnectionInfoReader conInfo(acceptorName, requesterName, acceptorRank, _addressDirectory);
+    std::string address = conInfo.read();
```

Comment:
Same here, therefore a stand-alone function may be a good idea to keep the code dry.